External/FEXCore/Source/Interface/Context/Context.h

```diff
@@ -337,6 +277,26 @@ namespace FEXCore::Context {
     // Public for threading
     void ExecutionThread(FEXCore::Core::InternalThreadState *Thread);
 
+    void FinalizeAOTIRCache() {
+      IRCaptureCache.FinalizeAOTIRCache();
+    }
+
+    void WriteFilesWithCode(std::function<void(const std::string& fileid, const std::string& filename)> Writer) {
+      IRCaptureCache.WriteFilesWithCode(Writer);
+    }
+
+    void SetAOTIRLoader(std::function<int(const std::string&)> CacheReader) {
+      IRCaptureCache.SetAOTIRLoader(CacheReader);
+    }
+
+    void SetAOTIRWriter(std::function<std::unique_ptr<std::ofstream>(const std::string&)> CacheWriter) {
+      IRCaptureCache.SetAOTIRWriter(CacheWriter);
+    }
+
+    void SetAOTIRRenamer(std::function<void(const std::string&)> CacheRenamer) {
```

Comment:
Any particular reason to use `std::function` here? It has heavy allocation overhead compared to a simple function pointer argument, which would work with the capture-less lambda used currently. (Whenever we do find a need to use capturing lambdas, we can just tune the interface after all)

Probably doesn't matter in this instance in particular, but it's an antipattern I've been observing in various places.

Reply:
This is purely to match behaviour with the above SetAOTIRWriter and SetAOTIRLoader.
This will change as I'm hammering on this code.